app/src/main/java/org/fossasia/phimpme/editor/editimage/EditImageActivity.java

```diff
@@ -355,10 +355,17 @@ public void changeMainBitmap(Bitmap newBit) {
 
     private void addToUndoList() {
         try{
+            TODO:// implement a more efficient way, like storing only the difference of bitmaps or
+            // steps followed to edit
+
             bitmapsForUndo.add(mainBitmap.copy(mainBitmap.getConfig(),true));
         }catch (OutOfMemoryError error){
-            //Snackbar.make(getWindow().getDecorView().getRootView(),"Out of Memory. steps = " + bitmapsForUndo.size(),Snackbar.LENGTH_LONG).show();
-
+            /**
+             * When outOfMemory exception throws then to make space, remove the last edited step
+             * from list and added the new operation in the end.
+             */
+            bitmapsForUndo.remove(0);
```

Comment:
remove the bitmap of index 1 from the list. 
because when we keep on undoing, it would be better if we end up with the original image rather than some randomly edited image.

I am not sure whether only just removing bitmap from the list would clear memory. I think you should call bitmap.recycle before removing that bitmap from the list.